pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -469,27 +471,65 @@ monitorLoop:
 				if err != nil {
 					logger.Reason(err).Error("failed to abort migration")
 				}
-				l.setMigrationResult(vmi, true, fmt.Sprintf("Live migration is not completed after %d sec and has been aborted", acceptableCompletionTime))
+				l.setMigrationResult(vmi, true, fmt.Sprintf("Live migration is not completed after %d sec and has been aborted", acceptableCompletionTime), v1.MigrationAbortSucceeded)
 				break monitorLoop
 			}
 
 		case libvirt.DOMAIN_JOB_NONE:
 			logger.Info("Migration job didn't start yet")
 		case libvirt.DOMAIN_JOB_COMPLETED:
-			logger.Info("Migration has beem completed")
+			logger.Info("Migration has been completed")
+			l.setMigrationResult(vmi, false, "", "")
 			break monitorLoop
 		case libvirt.DOMAIN_JOB_FAILED:
 			logger.Info("Migration job failed")
-			// migration failed
+			l.setMigrationResult(vmi, true, fmt.Sprintf("%v", err), "")
 			break monitorLoop
 		case libvirt.DOMAIN_JOB_CANCELLED:
 			logger.Info("Migration was canceled")
+			l.setMigrationResult(vmi, true, "Live migration aborted ", v1.MigrationAbortSucceeded)
 			break monitorLoop
 		}
 		time.Sleep(500 * time.Millisecond)
 	}
 }
 
+func (l *LibvirtDomainManager) CancelVMIMigration(vmi *v1.VirtualMachineInstance) error {
+	if vmi.Status.MigrationState == nil || vmi.Status.MigrationState.Completed ||
+		vmi.Status.MigrationState.Failed {
+
+		return fmt.Errorf("failed to cancel migration - vmi is not migrating")
```

Comment:
I wonder if it's technically possible for a migration to be marked for cancellation before the migration actually starts. I think that could actually happen if virt-handler wasn't able to react fast enough to start the migration (perhaps because of a rolling infrastructure update for example) before the migration object is deleted.


If that occurs, then we'll never get out of this loop. The cancel job will fail because there is no job and the work queue will keep processing that error and the migration would never start because it was marked for abortion. 

Do you think we should add a check here to ignore this function entirely if the MigrationState.StartTimestamp == nil?

Reply:
I don't know how can this happen, as virt-controller only requests to cancel if the `vmi.Status.MigrationState != nil` , also in asyncMigrationAbort() we cancel the migration only if it's in progress.
line 512

But, sure, I can add the additional check, no problem.